src/Scanner.js

```diff
@@ -279,6 +369,117 @@ class DefaultScanner {
+            '[class*=auth-content]',
+            '[class*=auth-modal]',
+            '[class*=auth-wrapper]',
+            '[class*=auth-stage]',
+            '[id*=auth-form]',
+            '[id*=auth-container]',
+            '[id*=auth-content]',
+            '[id*=auth-modal]',
+            '[id*=auth-wrapper]',
+            '[id*=authentication]',
+
+            // Login/signup specific patterns
+            '[class*=login-form]',
+            '[class*=signup-form]',
+            '[class*=sign-up-form]',
+            '[class*=registration-form]',
+            '[class*=register-form]',
+            '[id*=login-form]',
+            '[id*=login-container]',
+            '[id*=signup-form]',
+            '[id*=signup-container]',
+            '[id*=registration-form]',
+            '[id*=register-form]',
+            '[id*=registration]',
+            '[id*=register]',
+            '[id*=login]',
+            '[id*=signup]',
+
+            // Modal patterns
+            '[class*=modal-content]',
+            '[class*=modal-body]',
+            '[class*=modal-wrapper]',
+            '[class*=modal-container]',
+            '[class*=dialog-content]',
+            '[class*=dialog-body]',
+            '[id*=modal-content]',
+            '[id*=modal-body]',
+            '[id*=modal-wrapper]',
+            '[id*=modal-container]',
+            '[id*=dialog-content]',
+            '[id*=dialog-body]',
+            '[id*=modal]',
+            '[id*=dialog]',
+
+            // Form wrapper patterns
+            '[class*=form-wrapper]',
+            '[class*=form-container]',
+            '[class*=form-content]',
+            '[class*=form-body]',
+            '[class*=form-group]',
+            '[class*=form-row]',
+            '[id*=form-wrapper]',
+            '[id*=form-container]',
+            '[id*=form-content]',
+            '[id*=form-body]',
+            '[id*=form-group]',
+
+            // Generic form patterns
+            '[class*=form]:not([role=search])',
+            '[id*=form]:not([role=search])',
+
+            // Card patterns (common in modern UIs)
+            '[class*=card-body]',
+            '[class*=card-content]',
+            '[class*=card-container]',
+            '[id*=card-body]',
+            '[id*=card-content]',
+            '[id*=card-container]',
+
+            // Semantic HTML
+            '[role=form]',
+            '[role=dialog]',
+        ].join(',');
+
+        const formLikeContainer = input.closest(formSelectors);
+
+        // If we found a container, verify it's not too large and has a submit button
+        if (formLikeContainer) {
+            const containerInputs = formLikeContainer.querySelectorAll(this.matching.cssSelector('formInputsSelector'));
+            const submitButtons = formLikeContainer.querySelectorAll(this.matching.cssSelector('submitButtonSelector'));
+            if (containerInputs.length <= this.options.maxInputsPerForm && submitButtons.length > 0) {
```

Comment:
I think you might still need to check for `&& inputs.length > 1` to avoid getting trapped in something like `.class-form-row`.

```suggestion
            if (containerInputs.length <= this.options.maxInputsPerForm && inputs.length > 1 && submitButtons.length > 0) {
```

And you'll need recursion in that case. Adding the submit button requirement can also be revisited. Some forms might not have the submit button at all, or it can be some div.

But overall this seems directionally correct. Nice work!